sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitStmt.cs

```diff
@@ -1624,6 +1624,21 @@ private void VisitUnaryExprOrTypeTraitExpr(UnaryExprOrTypeTraitExpr unaryExprOrT
                         if (IsPrevContextStmt<CallExpr>(out var callExpr))
                         {
                             var index = callExpr.Args.IndexOf(unaryExprOrTypeTraitExpr);
+
+                            // If we didn't find the expression, try and find it under an implicit cast
+                            if (index == -1)
+                            {
+                                for (int i = 0; i < callExpr.Args.Count; i++)
```

Comment:
I wonder if it would be better to write this as a single loop:
```csharp
var args = callExpr.Args;
var index = -1;

for (int i = 0; i < args.Count; i++)
{
    var arg = args[i];

    if (IsStmtAsWritten(arg, unaryExprOrTypeTraitExpr))
    {
        index = i;
        break;
    }

    if (index == -1)
    {
        AddDiagnostic(DiagnosticLevel.Error, $"Failed to locate index of '{unaryExprOrTypeTraitExpr}' in callee declaration. Generated bindings may be incomplete.", calleeDecl);
    }
}
```

We can then have:
```csharp
private bool IsStmtAsWritten(Stmt stmt, Stmt expectedStmt, bool removeParens = false)
{
    if (stmt == expectedStmt)
    {
        return true;
    }

    if (stmt is not Expr expr)
    {
        return false;
    }

    expr = GetExprAsWritten(expr, removeParens);
    return expr == expectedStmt;
}
```

This will make it easier to cover more cases in the future and to consistently look into an expr to remove implicit casts or other bits as necessary.

Reply:
We have an existing `IsStmtAsWritten` that does a type check, rather than a value check, and it follows roughly this pattern.